tools/regconvert.py

```diff
@@ -11,13 +12,105 @@ def get_reg_macro(offset):
     reg = (offset - DATA_OFFSET) // 2
     group = reg // GROUP_SIZE
     reg_in_group = reg % GROUP_SIZE
-    return "%cREG(%d)\n" % (REGISTER_NAMES[group], reg_in_group)
+    return "%cREG(%d)" % (REGISTER_NAMES[group], reg_in_group)
+
+def parse_number(string, force_hex):
+    if string.lower().startswith("0x") or force_hex:
+        return int(string, 16)
+    return int(string, 10)
```

Comment:
If the hex string doesn't start with 0x and the user doesn't specify the hex arg, it'd be nice to still support automatically parsing a string like `A234`. You can change the second arg of `int()` to 0 and simplify it like this:

```
if force_hex:
    return int(string, 16)
return int(string, 0)
```